snapcraft/plugins/go.py

```diff
@@ -93,7 +93,10 @@ def get_pull_properties(cls):
 
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
-        self.build_packages.append("golang-go")
+        if ("go" not in self.build_snaps and
+            not any((s.startswith("go/") for s in self.build_snaps)) and
+            not any((s.startswith("go=") for s in self.build_snaps))):
```

Comment:
is this agreed upon with Gustavo? I'd rather introduce this later as you will need to fix our internal API for this to work too (snapcraft.internal.repo.snap) and also add a deprecation message for use of /